### PR TITLE
Fix plot legend issue #185

### LINKED ISSUE #185: Extra erroneous grobs in legend for `plot_demograph()`
Oddly, when one adds the legend to a plot_demograph(), and the fhx object lacks injuries, two round blobs appear in the legend. Thanks to Mannie Lopez @mlopez364 for pointing this out 

``` r
library(burnr)
library(dplyr)

data("pgm")

pgm_cl <- pgm %>% 
  filter(!rec_type %in% burnr:::rec_type_injury)

plot_demograph(pgm_cl,
               plot_legend = TRUE,
               composite_rug = TRUE,
               event_size = c(4, 3, 3),
               filter_prop = .10,
               filter_min_rec = 2,
               filter_min_events = 2)
```

<sup>Created on 2021-10-22 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

### PR DESCRIPTION
**Pull request by chguiterman**
This resolves a problem that (I think) originated in an update to {ggplot2}. If one of our aesthetic specifications was not present it was triggering {ggplot2} to build an extra legend grob. Adding `drop=FALSE` to `scale_shape_manual()` alleviated this problem and added flexibility to `plot_demograph()`

close #185

### CODE REVIEW

**Comment by brews** [comment]
I did a minor edit to the PR description so GH recognizes to automatically close #185.

**Review by brews — APPROVED** [review]
Thanks for the PR, @chguiterman 

I have one humble request for adding an in-line comment. Once you get that I think you can merge this whenever you're ready.

Thanks again!

File: R/utils.R [hunk 1]
```diff
@@ -412,8 +412,9 @@ get_rec_tbl <- function(x, injury_event = FALSE) {
   rec_tbl$series <- factor(rec_tbl$series,
                            levels = series_order,
                            ordered = TRUE)
-  rec_tbl$rec_type <- factor("recording")
-
+  if (dim(rec_tbl)[1] > 0) {
```
[the remark this answers is not in the document]
Comment (chguiterman) [reply R/utils.R:417]: No problem @brews. This change alleviated an error that was triggered if no trees were recording in the fhx object. Forcing the factor() on an empty datatset bonked. So the if() statement addresses that issue.

**Comment by brews** [comment]
Woohoo! Thanks for the update @chguiterman, I'm going to merge this.

**Pull request merged**